scipy/signal/_signaltools.py

```diff
@@ -4625,3 +4625,128 @@ def decimate(x, q, n=None, ftype='iir', axis=-1, zero_phase=True):
         sl[axis] = slice(None, None, q)
 
     return y[tuple(sl)]
+
+def envelope(x, N=None, method='analytic'):
```

Comment:
```suggestion
def envelope(x, *, N=None, method='analytic'):
```
Not sure what SciPy's policy is on keyword-only arguments, but I personally like making optional arguments keyword-only.